aiotask_context/__init__.py

```diff
@@ -1,23 +1,36 @@
 import asyncio
 import logging
+from copy import deepcopy
 
 
 logger = logging.getLogger(__name__)
 
 
-def task_factory(loop, coro):
+def _task_factory(loop, coro, copy_context):
```

Comment:
What about using a `kw` for the `copy_context` with the default value as `False` ?

Reply:
I believe this causes problems when you want to use the copying_task_context as the loop only calls the factory with the loop and coro parameters when the creates a task. It would be possible to use something like a partial to set the copy_context parameter whenever the loop calls the task factory but that doesn't feel too nice to me.

Reply:
When you register the task factory method you are registering the public method:

```python
loop.set_task_factory(context.copying_task_factory)
```

Asyncio loop will call this method, and internally the `copying_task_factory` will use the `kw`. Therefore, I can not see the problem that you are saying.

Reply:
I think I see what you mean. I originally thought you meant to remove the `copying_task_factory` and instead use a copy_context keyword argument only. I don't think this would work due to what I mentioned above, but keeping the copying_task_factory and using the keyword allows the new `task_factory` to be removed. I'll do that now